### PR TITLE
You won't believe what UI changes were made

### PR DESCRIPTION
**Pull request by walrys**
> What happens next might shock you...

# **User Dashboard UI fixes**
summarised below:
- moved several js style tags to separate LESS stylesheet for consistency
- made arrow appear only when tab is selected
- fix bug #329: tab selection does not switch to last tab on cancel add new planner
- fix bug #304: tab selection disappears when closing add new planner modal
- fix bug #330: delete modal hiding when clicking anywhere within modal except buttons
- implemented scrollbar for when tabs exceed length of container
- implemented right click to trigger dropdown menu
- fixed styling for dropdown menu
- made tab content scroll within container, so that tab container above remains visible and fixed
- refactored code and (jsx and stylesheets)

### CODE REVIEW

File: imports/ui/components/study_plan/Tab.jsx [hunk 1]
```diff
@@ -189,39 +188,71 @@ export default class Tab extends React.Component {
    */
   renderDropDownMenuSelection(className, iconClassName, iconStyle, text, clickHandler) {
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/Tab.jsx]: Multiple parameters are allowed to be passed in, which makes this component very reusable too.

File: imports/ui/components/study_plan/Tab.jsx [hunk 2]
```diff
@@ -189,39 +188,71 @@ export default class Tab extends React.Component {
...
-        </li>
+      <ReactClickOut onClickOut={this.handleOnBlurDropdown.bind(this)}>
+          <li className='nav-item'
+              onClick={this.props.onClickTab}
+              onContextMenu={onRightClick ? onRightClick : this.handleRightClick.bind(this)}
+              onMouseEnter={this.handleOnMouseEnter.bind(this)}
+              onMouseLeave={this.handleOnMouseLeave.bind(this)}>
+            <a className={this.setActiveTabClass('nav-link')} role='tab'>
+              <span className={this.props.navSpanClass} style={this.props.navSpanStyle}
+                  ref={(el)=> {
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/Tab.jsx]: Might need to comment about what's going on here/why there is an extra computation needed for the tab dimensions! Maybe even abstract this part into a helper function above with some code documentation. 

I'm guessing that it has to do with the position and width of the Tab. Small issue and we can discuss again tmr!

File: imports/ui/components/study_plan/Tab.less [hunk 3]
```diff
@@ -0,0 +1,50 @@
+// ========================================
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/Tab.less]: Awesome abstraction of styles from dumb components! We didn't think we'd need this at the beginning of our project but now, looking at this CSS stylesheet, it does indeed help a lot in maintaining CSS styles in the future! Next time, we can even use a React style selector that packages these CSS files into an object and we can import it into the React components that need that stylesheet, making it even more maintainable!

File: imports/ui/components/study_plan/Tab.less [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+//   Study Plan Tabs
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/Tab.less]: Oh and thanks for putting headings like these!

File: imports/ui/components/study_plan/TabDropdown.less [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  .nav-link-in .card-typical {
+    opacity: 1;
+
+    position: fixed;
+    top: 3em;
+    z-index: 300;
+    text-align: left;
+
+    background: #fff; /* For browsers that do not support gradients */
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/TabDropdown.less]: Good cross-browser support!

File: imports/ui/components/study_plan/TabDropdown.less [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+// ========================================
+
+// animate the dropdown menu
+  .keyframes(@name; @arguments) {
+  	@-moz-keyframes @name { @arguments(); }
+  	@-webkit-keyframes @name { @arguments(); }
+  	@keyframes @name { @arguments(); }
+  }
+
+  .animation(@arguments) {
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/TabDropdown.less]: Animations make a big difference in the 'feel' of the app!

File: imports/ui/components/study_plan/TabbedContainer.jsx [hunk 7]
```diff
@@ -195,23 +215,24 @@ export default class TabbedContainer extends React.Component {
...
     } else {
-      tabTitleComponent = this.renderTabTitle(tabTitle, plannerID)
+      tabTitleComponent = this.renderTabTitle(tabTitle, plannerID, isActiveTab);
     }
 
     return (
-      <Tab key={index}
-           navSpanClass="nav-link-in" navSpanStyle={{position: "relative"}}
-           tabTitle={tabTitleComponent}
+      <Tab key={index} ref={isActiveTab ? this.scrollToElement : null}
```
Comment (NatashaKSS) [inline imports/ui/components/study_plan/TabbedContainer.jsx]: Ahh I see! Now I see how this scrollToElement is used!

File: imports/ui/components/study_plan/Tab.jsx (hunk 1 above)
Comment (walrys) [inline imports/ui/components/study_plan/Tab.jsx]: LOL I believe this part is your code Natasha

**Pull request merged**